Update release note (#37)
keizer619/release-notes: release-notes-1.0.0.md

## Language

- A set of modules, which contain functions associated with the basic types have been introduced. Collectively, these modules are referred to as the lang library. Each basic type has a corresponding lang library module. Additionally, there is also the `lang.value` module, which holds functions common for all the types. The following is the list of lang library modules.

  - `ballerina/lang.value`

>   - `ballerina/lang.array` for basic type list

Reviewer: ```suggestion
  - `ballerina/lang.array` for list types
```

  - `ballerina/lang.float` for basic type `float`
  - `ballerina/lang.future` for basic type `future`
  - `ballerina/lang.int` for basic type `int`

>   - `ballerina/lang.map` for basic type mapping

Reviewer: ```suggestion
  - `ballerina/lang.map` for mapping types
```

- Revamp the NATS connector to support both NATS and Streaming Servers.
- Introduce the standard library module-wise errors as a replacement for the builtin `error`.
  e.g., Ballerina HTTP Error types include `http:ClientError`, `http:ListenerError`, `http:ClientAuthError` etc.
- Introduce capability to engage custom providers and handlers for inbound/outbound authentication.
- Introduce OAuth2 inbound authentication.
- Introduce own modules for different authentication mechanisms (JWT, LDAP, OAuth2 etc.).
- Introduce prior knowledge support to the HTTP/2 client.
- Add flow control support to HTTP/2 client and server.
- Introduce XSLT transformation support.
- `ballerina/h2` and `ballerina/mysql` database client modules and the `ballerina/sql` module have been discontinued. The `ballerinax/java.jdbc` client module can be used to interact with relational databases.

> - The `ByteChannel` read API was updated to return only `byte[]|io:Error`, removing the previously returned read byte length.

Author: This sentence still sounds weird. Any ideas on how this can be improved? @MaryamZi

Reviewer: ```suggestion
- The byte channel read API was updated to return only `byte[]|io:Error`.
```
